pkg/dependencies/factory.go

```diff
@@ -846,7 +883,12 @@ func (f *Factory) WithClusterManager(clusterConfig *v1alpha1.Cluster, opts ...cl
 			},
 			clustermanager.DefaultRetrier(),
 		)
-		installer := clustermanager.NewEKSAInstaller(client, f.dependencies.FileReader)
+
+		if timeoutOpts != nil && timeoutOpts.NoTimeout {
+			client.Retrier = retrier.NewWithNoTimeout()
```

Comment:
nit: it would be cleaner if just pass the right retrier to the constructor
And ideally make `client.Retrier` -> `client.retrier`

Reply:
i will do the client.Retrier -> retrier in separate PR